src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -35,15 +37,26 @@ public JsonSchemaTransformer(JsonSchema schema)
 
             void TransformXrefScalar(JsonSchema schema, JValue value)
             {
-                extensions[value.Path] = new Lazy<JValue>(
-                    () => TransformScalar(schema, file, context, value, errors, buildChild: null),
-                    LazyThreadSafetyMode.PublicationOnly);
+                var name = value.Path;
+                if (!string.IsNullOrEmpty(value.Parent?.Parent?.Path))
+                {
+                    Debug.Assert(name.StartsWith(value.Parent.Parent.Path));
+                    name = name.Substring(value.Parent.Parent.Path.Length + 1).Trim(new char[] { '\'', ']', '[' });
+                }
+                if (schema.Parent?.XrefProperties.Contains(name) ?? false)
+                {
+                    extensions[value.Path] = new Lazy<JValue>(
+                        () => TransformScalar(schema, file, context, value, errors, buildChild: null),
+                        LazyThreadSafetyMode.PublicationOnly);
+                }
             }
         }
 
         private void Traverse(JsonSchema schema, JToken token, Action<JsonSchema, JValue> transformScalar)
```

Comment:
Use a property in a data model with `[JsonIgnore]` to pass parameters is __NOT__ thread safe. The same JsonSchema instance is shared by multiple concurrent calls to Traverse.

```suggestion
        private void Traverse(JsonSchema schema, JsonSchema parentSchema, JToken token, Action<JsonSchema, JsonSchema, JValue> transformScalar)
```

Reply:
OK, but can we make it not shared with multiple concurrent calls just like the JToken passed in?